mapper/orm/src/main/java/org/hibernate/search/mapper/orm/scope/SearchScope.java

```diff
@@ -38,7 +38,8 @@
  *
  * @param <E> A supertype of all types in this scope.
  */
-public interface SearchScope<E> {
+@SuppressWarnings("deprecation")
+public interface SearchScope<E> extends org.hibernate.search.engine.mapper.scope.SearchScope<E, org.hibernate.search.mapper.orm.common.EntityReference> {
```

Comment:
Obviously, JQAssistant does not like these changes 😔 😃. However, I couldn't come up with an alternative to allow scope to be created from the "generated" static metamodel classes. Unless we generate a model specific to the mapper 😖 😕 

@yrodiere if you can take a quick look at the approach in this PR whenever you have some free time 😃

Reply:
> Unless we generate a model specific to the mapper 😖 😕

That seems like a very valid solution to me?

Not for every Hibernate Search type in the metamodel, of course -- i.e. I wouldn't make `ValueFieldReference` specific to each mapper. But it seems perfectly fine to me to give `EntityA_.scope` a mapper-specific type.

Reply:
> But it seems perfectly fine to me to give EntityA_.scope a mapper-specific type.

mm ok, we'd need to tell the plugin generating these to know which mappers are expected and then have 
```java
EntityA_.ormScope
EntityA_.standaloneScope
```
we probably can detect that from the dependencies (which mappers are present) and have an explicit property if needed I suppose 🤔

Reply:
> mm ok, we'd need to tell the plugin generating these to know which mappers are expected and then have

The plugin would already need to know, since it needs to start Hibernate Search in offline mode to create backends and their index descriptors.